main.js

```diff
@@ -0,0 +1,123 @@
+//make a search button that fetches the country data when you click it
+const searchButton = document.getElementById("search-button");
+
+searchButton.addEventListener('click', (event) => {
+    fetchData();
+});
+
+//make a search bar that fetches the country data when pressing enter
+const searchBar = document.getElementById('input-field');
+searchBar.addEventListener('keyup', setQuery)
+
+let query = '';
+
+function setQuery(e) {
+    query = e.target.value;
+    if (e.keyCode === 13) {
+        fetchData();
+    }
+}
+
+//make a container for all the info about the country
+const countryContainer = document.getElementById("countryInformation")
+
+//asynchronous function for getting data about the country
+async function fetchData() {
+    searchBar.value = "";
+    //make sure the previous result gets cleared
+    // ????
+    const previousResult = document.getElementById("countryAll");
+    if (previousResult) {
+        countryContainer.removeChild(previousResult);
+    };
+    try {
+        //GET request
+        const result = await axios.get(`https://restcountries.eu/rest/v2/name/${query}`);
```

Comment:
Wat ik zelf heel vervelend vind is dat ik heel veel `result.data[0]` moet herhalen. Wij developers zijn heel lui, dus is daarom de object spreader in het leven geroepen 😉 
```
const { name: countryName, population, subregion: sbuAreaName, ...otherProps } = result.data[0];
```

_Je hoeft de props niet perse te aliasen_